File: php_SDK/Samples/RecurringBillStatusSample.php
<?php
/* The current url, oauth_token, orgId and Terminal Id provided in this example, are only for testing purposes
*  For development purposes you need to contact the Payhub Integration Support team. They will provide you with  *   all you need.
*  Thanks.
*/
$path_to_IncludeClases="../com/payhub/ws/extra/includeClasses.php";
include_once $path_to_IncludeClases;

//Defining the Web Service URL
$WsURL="https://sandbox-api.payhub.com/api/v2/";
$oauth_token = "2a5d6a73-d294-4fba-bfba-957a4948d4a3";

//Defining data for the SALE transaction
// Merchant data (obtained from the payHub Virtual Terminal (3rd party integration)
$merchant = new Merchant();
$merchant->setOrganizationId(10074);
$merchant->setTerminalId(134);

$bill= new Bill();
$bill->setBaseAmount(1.00);
//Credit card data
$card_data = new CardData();
$card_data->setCardNumber("4012881888818888");
$card_data->setCardExpiryDate("202012");
$card_data->setBillingZip("60527");
// Customer data
$customer = new Customer();
$customer->setFirstName("Joe");
$customer->setLastName("Tester");
$customer->setPhoneNumber("844-217-1631");
$customer->setPhoneType("M");

$montly_s = new MontlySchedule();
$montly_s->monthly_type="E";
$montly_s->monthly_each_days=array(15);

$start=new DateTime("2015-9-29");
$start=$start->format('Y-m-d');
$type="O";
$endDate=new DateTime("2016-8-29");
$endDate=$endDate->format('Y-m-d');
$scheduleSandE=new ScheduleSartAndEnd();
$scheduleSandE->start_date=$start;
$scheduleSandE->end_date_type=$type;
$scheduleSandE->end_date=$endDate;

$schedule =new Schedule($scheduleSandE,$montly_s);

$schedule->schedule_type="M";
$schedule->bill_generation_interval=1;

$recurringBill=new RecurringBill($merchant,$customer,$bill,$card_data,$schedule);
$transaction = new TransactionManager($merchant,$WsURL,$oauth_token);
$response = $transaction->doRecurringBill($recurringBill);

if($response->getErrors()==null){
    $transactionId = $response->getLastRecurringBillResponse()->getRecurringBillId();
    $statusInfo = $response->getStatusInformation();
    var_dump($statusInfo);
    echo "Updating...";
    $canUpdate=$transaction->updateRecurringBillStatus($transactionId);
    if($canUpdate){
        $statusUpdated = $transaction->getRecurringBillInformation($transactionId);
        if($statusUpdated->getErrors()==null){
            var_dump($statusUpdated->getStatusInformation());
        }
    }
}else{
    var_dump($response->getErrors());
}

File: java_SDK/Samples/RecurringBillStatusSample.java
package Samples;

import java.io.IOException;
import java.math.BigDecimal;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

import com.payhub.ws.api.RecurringBillResponseInformation;
import com.payhub.ws.api.TransactionManager;
import com.payhub.ws.model.Bill;
import com.payhub.ws.model.CardData;
import com.payhub.ws.model.Customer;
import com.payhub.ws.model.Merchant;
import com.payhub.ws.model.MontlySchedule;
import com.payhub.ws.model.RecurringBill;
import com.payhub.ws.model.Schedule;
import com.payhub.ws.model.ScheduleSartAndEnd;
import com.payhub.ws.model.TransactionAmount;


public class RecurringBillStatusSample {

	public void updateRecurringBill() throws IOException, ParseException {
		
	        String url = "https://staging-api.payhub.com/api/v2/";
	        String oauth = "bb96358e-2aa8-4c6c-8a2e-901b676e979d";
	        ScheduleSartAndEnd scheduleSartAndEnd = new ScheduleSartAndEnd();
	        scheduleSartAndEnd.setStartDate("2015-09-20");
	        scheduleSartAndEnd.setEndDate("2016-07-08");
	        scheduleSartAndEnd.setEnd_date_type("O");
	        MontlySchedule montlySchedule = new MontlySchedule();
	        montlySchedule.setMonthly_type("E");
	        List<Integer> m = new ArrayList<Integer>();
	        m.add(15);
	        montlySchedule.setMonthly_each_days(m);
	        Schedule schedule = new Schedule(scheduleSartAndEnd, montlySchedule);
	        schedule.setSchedule_type("M");
	        schedule.setBill_generation_interval(1);
	        
	        Merchant merchant = new Merchant();
	        merchant.setOrganization_id(10002);
	        merchant.setTerminal_id(2);
	        
	        Bill bill = new Bill();
	        bill.setBase_amount(new TransactionAmount().dollars(new BigDecimal(100)));
	        
	        CardData card_data = new CardData();
	        card_data.setCard_number( "4012881888818888");
	        card_data.setCard_expiry_date("202012"); //September 2018
	        card_data.setBilling_zip ("60527");
	        
	        Customer customer = new Customer();
	        customer.setFirst_name("Joe");
	        customer.setLast_name("Tester");
	        customer.setPhone_number("844-217-1631");
	        customer.setPhone_type("W");
	        
	        RecurringBill recurringBill = new RecurringBill(merchant,card_data,customer,schedule,bill);
	        
	        TransactionManager transaction = new TransactionManager(url, oauth, merchant);
	        RecurringBillResponseInformation response = transaction.doRecurringBill(recurringBill);
	        if(response.getErrors()==null){
	    		String id = response.getLastRecurringBillResponse().getRecurringBillId();
	    		System.out.println(response.getStatusInformation().getStatus().getRecurring_bill_status());
	    		boolean canUpdate = transaction.updateRecurringBillStatus(id);
	    		if(canUpdate){
	    			RecurringBillResponseInformation responseUpdated = transaction.getRecurringBillInformation(id);
	    			if(responseUpdated.getErrors()==null){
	    				System.out.println(responseUpdated.getStatusInformation().getStatus().getRecurring_bill_status());
	    			}
	    		}
	        }
	}

}
